src/main/java/sparksoniq/jsoniq/item/DurationItem.java

```diff
@@ -0,0 +1,183 @@
+package sparksoniq.jsoniq.item;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.joda.time.Instant;
+import org.joda.time.Period;
+import org.joda.time.PeriodType;
+import org.joda.time.format.ISOPeriodFormat;
+import org.joda.time.format.PeriodFormatter;
+import org.rumbledb.api.Item;
+import sparksoniq.exceptions.IteratorFlowException;
+import sparksoniq.exceptions.UnexpectedTypeException;
+import sparksoniq.jsoniq.compiler.translator.expr.operational.base.OperationalExpressionBase;
+import sparksoniq.jsoniq.runtime.metadata.IteratorMetadata;
+import sparksoniq.semantics.types.AtomicTypes;
+import sparksoniq.semantics.types.ItemType;
+import sparksoniq.semantics.types.ItemTypes;
+
+import java.util.regex.Pattern;
+
+public class DurationItem extends AtomicItem {
+
+    private static final String prefix = "(-)?P";
+    private static final String duYearFrag = "(\\d)+Y";
+    private static final String duMonthFrag = "(\\d)+M";
+    private static final String duDayFrag = "(\\d)+D";
+    private static final String duHourFrag = "(\\d)+H";
+    private static final String duMinuteFrag = "(\\d)+M";
+    private static final String duSecondFrag = "(((\\d)+)|(\\.(\\d)+)|((\\d)+\\.(\\d)+))S";
+
+    private static final String duYearMonthFrag = "((" + duYearFrag + "(" + duMonthFrag + ")?)|" + duMonthFrag + ")";
+    private static final String duTimeFrag = "T((" + duHourFrag + "(" + duMinuteFrag + ")?" + "(" + duSecondFrag + ")?)|" +
+            "(" + duMinuteFrag + "(" + duSecondFrag + ")?)|" + duSecondFrag + ")";
+    private static final String duDayTimeFrag = "((" + duDayFrag + "(" + duTimeFrag + ")?)|" + duTimeFrag + ")";
+    private static final String durationLiteral = prefix + "((" + duYearMonthFrag + "(" + duDayTimeFrag + ")?)|" + duDayTimeFrag +")";
+    private static final Pattern pattern = Pattern.compile(durationLiteral);
+
+
+    private static final long serialVersionUID = 1L;
+    private Period _value;
+
+    public DurationItem() {
+        super();
+    }
+
+    public DurationItem(Period value) {
+        super();
+        this._value = value.normalizedStandard(PeriodType.yearMonthDayTime());
+    }
+
+    public Period getValue() {
```

Comment:
It would be nice to expose this function in the Item class, renamed as getDurationValue(), as is done for all others (getIntegerValue(), getStringValue(), getBooleanValue(), etc...). As for the other getXXXValue() functions, Item.getDurationValue should throw an exception and DurationItem.getDurationValue overrides it here.

That way, this is consistent with the rest of the codebase, there is no need to cast to DurationItem (as is done at three places down there) to obtain the period, and users of the public Item API will be able to extract the period from the item if it is a duration, in the same way they extract other types of values.

Reply:
This sounds like a good solution. Thanks!